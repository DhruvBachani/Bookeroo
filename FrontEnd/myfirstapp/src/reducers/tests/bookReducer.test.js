import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import { initialState } from '../bookReducer';

describe('bookReducer', () => {
    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    })

    it('should handle GET_BOOKS', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_BOOKS'
                })
        ).toMatchSnapshot();
    })

    it('should handle GET_CATEGORIES', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_CATEGORIES'
                })
        ).toMatchSnapshot();
    })

    it('should handle GET_BOOK', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_BOOK'
                })
        ).toMatchSnapshot();
    })
})