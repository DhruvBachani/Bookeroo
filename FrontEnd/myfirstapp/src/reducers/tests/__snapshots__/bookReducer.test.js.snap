// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookReducer should handle GET_BOOK 1`] = `
Object {
  "book": Object {
    "book": undefined,
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`bookReducer should handle GET_BOOKS 1`] = `
Object {
  "book": Object {
    "books": undefined,
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`bookReducer should handle GET_CATEGORIES 1`] = `
Object {
  "book": Object {
    "allCategories": undefined,
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`bookReducer should return the initial state 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;
